### PR TITLE
feat(plasma-typo,plasma-ui): Add breakWord props for global style root

### PR DESCRIPTION
**Pull request by neretin-trike**
В данной пул-реквесте реализовал возможность глобального отключения переносов для новой типографики.
Чтобы переопределить стандартное поведение компонентов типографики, нужно указать `breakWord={false}` в компоненте `DeviceThemeProvider`, или у компонента глобального стиля:

Для plasma-b2c / web
```ts
import { standard as standardTypo, compatible as compatibleTypo } from '@salutejs/plasma-typo';

const TypoStyle = createGlobalStyle(standardTypo);

<>
    <TypoStyle breakWord={false} />
    <CompatibleTypoStyle />
</>
```

Для plasma-ui
```ts
import { DeviceThemeProvider } from '@salutejs/plasma-ui';

const TypoStyle = createGlobalStyle(standardTypo);

<>
    <DeviceThemeProvider breakWord={false}>
    ....
    </DeviceThemeProvider>
</>

```

:baby_chick: Download canary assets:

[colors--canary.310.4113711817.xml](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/colors--canary.310.4113711817.xml)  
[color_metro_ios-swift--canary.310.4113711817.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_metro_ios-swift--canary.310.4113711817.swift)  
[color_metro_kotlin--canary.310.4113711817.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_metro_kotlin--canary.310.4113711817.kt)  
[color_metro_react-native--canary.310.4113711817.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_metro_react-native--canary.310.4113711817.ts)  
[color_sbermarket_ios-swift--canary.310.4113711817.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_ios-swift--canary.310.4113711817.swift)  
[color_sbermarket_kotlin--canary.310.4113711817.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_kotlin--canary.310.4113711817.kt)  
[color_sbermarket_react-native--canary.310.4113711817.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_react-native--canary.310.4113711817.ts)  
[color_sberprime_ios-swift--canary.310.4113711817.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_ios-swift--canary.310.4113711817.swift)  
[color_sberprime_kotlin--canary.310.4113711817.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_kotlin--canary.310.4113711817.kt)  
[color_sberprime_react-native--canary.310.4113711817.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_react-native--canary.310.4113711817.ts)  
[color_selgros_ios-swift--canary.310.4113711817.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_selgros_ios-swift--canary.310.4113711817.swift)  
[color_selgros_kotlin--canary.310.4113711817.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_selgros_kotlin--canary.310.4113711817.kt)  
[color_selgros_react-native--canary.310.4113711817.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_selgros_react-native--canary.310.4113711817.ts)  
[color_smbusiness_ios-swift--canary.310.4113711817.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_smbusiness_ios-swift--canary.310.4113711817.swift)  
[color_smbusiness_kotlin--canary.310.4113711817.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_smbusiness_kotlin--canary.310.4113711817.kt)  
[color_smbusiness_react-native--canary.310.4113711817.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_smbusiness_react-native--canary.310.4113711817.ts)  
[PlasmaTokensColor--canary.310.4113711817.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/PlasmaTokensColor--canary.310.4113711817.swift)  
[shadow_sbermarket_react-native--canary.310.4113711817.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/shadow_sbermarket_react-native--canary.310.4113711817.ts)  
[typo_mage_ios-swift--canary.310.4113711817.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_ios-swift--canary.310.4113711817.swift)  
[typo_mage_kotlin--canary.310.4113711817.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_kotlin--canary.310.4113711817.kt)  
[typo_mage_react-native--canary.310.4113711817.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_react-native--canary.310.4113711817.ts)  
[typo_plasma_ios-swift--canary.310.4113711817.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_ios-swift--canary.310.4113711817.swift)  
[typo_plasma_kotlin--canary.310.4113711817.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_kotlin--canary.310.4113711817.kt)  
[typo_plasma_react-native--canary.310.4113711817.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_react-native--canary.310.4113711817.ts)  
[typo_ruler_ios-swift--canary.310.4113711817.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_ios-swift--canary.310.4113711817.swift)  
[typo_ruler_kotlin--canary.310.4113711817.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_kotlin--canary.310.4113711817.kt)  
[typo_ruler_react-native--canary.310.4113711817.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_react-native--canary.310.4113711817.ts)  
[typo_sage_ios-swift--canary.310.4113711817.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_ios-swift--canary.310.4113711817.swift)  
[typo_sage_kotlin--canary.310.4113711817.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_kotlin--canary.310.4113711817.kt)  
[typo_sage_react-native--canary.310.4113711817.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_react-native--canary.310.4113711817.ts)  
[typo_sbermarket_ios-swift--canary.310.4113711817.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_ios-swift--canary.310.4113711817.swift)  
[typo_sbermarket_kotlin--canary.310.4113711817.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_kotlin--canary.310.4113711817.kt)  
[typo_sbermarket_react-native--canary.310.4113711817.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_react-native--canary.310.4113711817.ts)  
[typo_soulmate_ios-swift--canary.310.4113711817.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_ios-swift--canary.310.4113711817.swift)  
[typo_soulmate_kotlin--canary.310.4113711817.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_kotlin--canary.310.4113711817.kt)  
[typo_soulmate_react-native--canary.310.4113711817.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_react-native--canary.310.4113711817.ts)

<details>
  <summary>📦 Published PR as canary version: <code>Canary Versions</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @salutejs/plasma-b2c@1.135.0-canary.310.4113711817.0
  npm install @salutejs/plasma-core@1.93.0-canary.310.4113711817.0
  npm install @salutejs/plasma-icons@1.121.0-canary.310.4113711817.0
  npm install @salutejs/plasma-temple@1.130.0-canary.310.4113711817.0
  npm install @salutejs/plasma-tokens-b2b@1.8.0-canary.310.4113711817.0
  npm install @salutejs/plasma-tokens-b2c@0.17.0-canary.310.4113711817.0
  npm install @salutejs/plasma-tokens-web@1.23.0-canary.310.4113711817.0
  npm install @salutejs/plasma-tokens@1.34.0-canary.310.4113711817.0
  npm install @salutejs/plasma-typo@0.30.0-canary.310.4113711817.0
  npm install @salutejs/plasma-ui@1.163.0-canary.310.4113711817.0
  npm install @salutejs/plasma-web@1.161.0-canary.310.4113711817.0
  npm install @salutejs/plasma-cy-utils@0.41.0-canary.310.4113711817.0
  npm install @salutejs/plasma-sb-utils@0.91.0-canary.310.4113711817.0
  npm install @salutejs/plasma-tokens-android@2.41.0-canary.310.4113711817.0
  npm install @salutejs/plasma-tokens-ios-swift@2.41.0-canary.310.4113711817.0
  npm install @salutejs/plasma-tokens-utils@0.16.0-canary.310.4113711817.0
  # or 
  yarn add @salutejs/plasma-b2c@1.135.0-canary.310.4113711817.0
  yarn add @salutejs/plasma-core@1.93.0-canary.310.4113711817.0
  yarn add @salutejs/plasma-icons@1.121.0-canary.310.4113711817.0
  yarn add @salutejs/plasma-temple@1.130.0-canary.310.4113711817.0
  yarn add @salutejs/plasma-tokens-b2b@1.8.0-canary.310.4113711817.0
  yarn add @salutejs/plasma-tokens-b2c@0.17.0-canary.310.4113711817.0
  yarn add @salutejs/plasma-tokens-web@1.23.0-canary.310.4113711817.0
  yarn add @salutejs/plasma-tokens@1.34.0-canary.310.4113711817.0
  yarn add @salutejs/plasma-typo@0.30.0-canary.310.4113711817.0
  yarn add @salutejs/plasma-ui@1.163.0-canary.310.4113711817.0
  yarn add @salutejs/plasma-web@1.161.0-canary.310.4113711817.0
  yarn add @salutejs/plasma-cy-utils@0.41.0-canary.310.4113711817.0
  yarn add @salutejs/plasma-sb-utils@0.91.0-canary.310.4113711817.0
  yarn add @salutejs/plasma-tokens-android@2.41.0-canary.310.4113711817.0
  yarn add @salutejs/plasma-tokens-ios-swift@2.41.0-canary.310.4113711817.0
  yarn add @salutejs/plasma-tokens-utils@0.16.0-canary.310.4113711817.0
  ```
</details>

### CODE REVIEW

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-310/

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **🌚 Missing base report**

Check out current report in job artifacts!

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/pr/demo-canvas-app-pr-310/

**Comment by Yeti-or** [comment]
Описать использование

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-310/

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **💀 Missing report, something went wrong**

Check out current report in job artifacts!

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/pr/demo-canvas-app-pr-310/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr/pr-310/
  ui storybook: http://plasma.sberdevices.ru/pr/pr-310/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr/pr-310/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr/pr-310/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr/pr-310/temple-storybook/

File: website/plasma-ui-docs/docs/components/Typography.mdx [hunk 1]
```diff
@@ -17,7 +17,7 @@ import { StorybookLink } from '@site/src/components';
 По умолчанию типографика поддерживает стандартные отступы. О том, как использовать отступы, читайте на странице [Стандартные отступы](../design/spacing#встроенные-отступы-в-типографике).
 
 ## Переносы
-По умолчанию в типографике слова переносятся по слогам. Чтобы отключить данный функционал, необходимо установить свойство `breakWord` в значении `false`.
+По умолчанию в типографике слова переносятся по слогам. Чтобы отключить данный функционал, необходимо установить свойство `breakWord` в значении `false`. Если свойство `breakWord` определено на уровне компонента `DeviceThemeProvider` в значении `false`, то добавление данного пропса на элемент типографики не изменит его поведение.
```
Comment (Yeti-or) [inline website/plasma-ui-docs/docs/components/Typography.mdx:20]: стоп кажется это не верно, если я глобально типографику отключил, но хочу в каком-то компоненте её таки включить?

Comment (neretin-trike) [reply website/plasma-ui-docs/docs/components/Typography.mdx:20]: понимаю о чём ты, и думал, как это можно сделать. Но с текущим API я не придумал, как этого добиться без прокидывания дополнительного контекста

Comment (Yeti-or) [reply website/plasma-ui-docs/docs/components/Typography.mdx:20]: Пускай по умолчанию в теме задается 

```css
break-word: = true => 

--plasma-typo-hyphens: auto


# тоже самое в компонентах через appyHyphens()
css:
   hyphens: var(--plasma-typo-hyphens)

# break-word тогда по умолчанию не задан,
# а если задан то меняет сам токен
--plasma-typo-hyphens: break-word  ?  auto: manual;

```

Comment (neretin-trike) [reply website/plasma-ui-docs/docs/components/Typography.mdx:20]: а вообще в итоге получлиось, через предопределение кастомных переменных

Comment (neretin-trike) [reply website/plasma-ui-docs/docs/components/Typography.mdx:20]: о, не видел твоего сообщения :D

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-310/

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **💀 Missing report, something went wrong**

Check out current report in job artifacts!

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/pr/demo-canvas-app-pr-310/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr/pr-310/
  ui storybook: http://plasma.sberdevices.ru/pr/pr-310/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr/pr-310/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr/pr-310/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr/pr-310/temple-storybook/

File: website/plasma-ui-docs/docs/components/Typography.mdx (hunk 1 above)
Comment (Yeti-or) [reply website/plasma-ui-docs/docs/components/Typography.mdx]: а сломается ли оно если использовать со старой типографикой?

**Review by Yakutoc — APPROVED** [review]

**Review by Yeti-or — APPROVED** [review]

**Pull request merged**